src/test/java/com/microsoft/sqlserver/jdbc/connection/PoolingTest.java

```diff
@@ -139,4 +145,52 @@ public void testConnectionPoolClientConnectionId() throws SQLException {
 
         assertEquals(Id1, Id2, "ClientConnection Ids from pool are not the same.");
     }
+    
+    @Test
+    public void testHikariCP() throws SQLException {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(connectionString);
+        HikariDataSource ds = new HikariDataSource(config);
+
+        connect(ds);
+    }
+
+    @Test
+    public void testApacheDBCP() throws SQLException {
+        BasicDataSource ds = new BasicDataSource();
+        ds.setUrl(connectionString);
+
+        connect(ds);
+    }
+
+    private static void connect(DataSource ds) throws SQLException {
+        Connection con = null;
+        PreparedStatement pst = null;
+        ResultSet rs = null;
+
+        try {
+            con = ds.getConnection();
+            con.isValid(5);
+
+            pst = con.prepareStatement("SELECT SUSER_SNAME()");
```

Comment:
Currently "isValid" ends up using the timeout timer, but this is an implementation choice (other JDBC drivers do it differently for speed and this JDBC driver might one day too) and might change in the future. Also "isValid" is already called by the connection pool before returing a connection from the pool (see for example "validationTimeout" in [HikariCP](https://github.com/brettwooldridge/HikariCP/)).

Besides that, the original issue was about "queries with a time-out", so why not test that explicitly and make sure to use a query with a time-out?